nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/impl/MSSQL2008DatabaseAdapter.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard.db.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.processors.standard.db.DatabaseAdapter;
+
+/**
+ * A database adapter that generates MS SQL Compatible SQL for version 2008.
+ */
+public class MSSQL2008DatabaseAdapter implements DatabaseAdapter {
+    @Override
+    public String getName() {
+        return "MS SQL 2008";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Generates MS SQL Compatible SQL for version 2008";
+    }
+
+    @Override
+    public String getSelectStatement(String tableName, String columnNames, String whereClause, String orderByClause, Long limit, Long offset) {
+        if (StringUtils.isEmpty(tableName)) {
+            throw new IllegalArgumentException("Table name cannot be null or empty");
+        }
+
+        final StringBuilder query = new StringBuilder("SELECT ");
+
+        // If this is a limit query and not a paging query then use TOP in MS SQL
+        if (limit != null) {
+
+            if (offset != null) {
+                query.append("* FROM (SELECT ");
+            }
+            final long effectiveOffset = (offset == null) ? 0 : offset;
+            if (effectiveOffset + limit > 0) {
+                query.append("TOP ");
+                query.append(effectiveOffset + limit);
+                query.append(" ");
+            }
+        }
+
+        if (StringUtils.isEmpty(columnNames) || columnNames.trim().equals("*")) {
+            query.append("*");
+        } else {
+            query.append(columnNames);
+        }
+
+        if (limit != null && offset != null) {
+            query.append(", ROW_NUMBER() OVER(ORDER BY ID asc) rnum");
```

Comment:
@mattyb149 I don't think there's a guarantee that the `ORDBER BY` column is `ID`. You meant for this to be `orderByClause`, right?